configs/components/_base-ruby.rb

```diff
@@ -48,7 +48,8 @@
 elsif platform.is_macos?
   pkg.environment 'optflags', settings[:cflags]
   if platform.is_cross_compiled?
-    pkg.build_requires "ruby@#{ruby_version_y}"
+    # Pin to an older version of ruby@2.5. This can be removed once we're no longer cross-compiling
+    pkg.build_requires "joshcooper/puppet/ruby@#{ruby_version_y}"
```

Comment:
We only want to do this for projects that rely on `ruby@2.5` like `agent-runtime-6.x` but not `agent-runtime-main`.